components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/template/APITemplateBuilderImpl.java

```diff
@@ -41,65 +49,100 @@
+		configcontext = new HandlerConfigContex(configcontext, handlers);
+		configcontext = new EnvironmentConfigContext(configcontext, environment);
+		configcontext = new TemplateUtilContext(configcontext);
+
+		// @todo: this validation might be better to do when the builder is initialized.
 		try {
-			// build the context for template and apply the necessary decorators
-			ConfigContext configcontext = new APIConfigContext(this.api);
-
-			configcontext = new TransportConfigContext(configcontext, api);
-			configcontext = new ResourceConfigContext(configcontext, api);
-			// this should be initialised before endpoint config context.
-			// configcontext = new EndpointBckConfigContext(configcontext,api);
-			// configcontext = new EndpointConfigContext(configcontext, api);
-			configcontext = new EndpointURIConfigContext(configcontext, api);
-			configcontext = new SecurityConfigContext(configcontext, api);
-			configcontext = new JwtConfigContext(configcontext);
-			configcontext = new ResponseCacheConfigContext(configcontext, api);
-			configcontext = new HandlerConfigContex(configcontext, handlers);
-			configcontext = new EnvironmentConfigContext(configcontext, environment);
-			configcontext = new TemplateUtilContext(configcontext);
-
-			// @todo: this validation might be better to do when the builder is
-			// initialized.
 			configcontext.validate();
+		} catch (Exception e) {
+			log.error("Cannot validate configuration context for template", e);
+			throw new APITemplateException("Cannot validate configuration context for template",
+										   e);
+		}
 
-			VelocityContext context = configcontext.getContext();
-
-			/* first, initialize velocity engine */
-			VelocityEngine velocityengine = new VelocityEngine();
+		VelocityEngine velocityengine = new VelocityEngine();
+		if (this.velocityLoggerName != null) {
+			velocityengine.setProperty(RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+									   "org.apache.velocity.runtime.log.Log4JLogChute");
+			velocityengine.setProperty("runtime.log.logsystem.log4j.logger", velocityLoggerName);
+		}
+		try {
 			velocityengine.init();
+		} catch (Exception e) {
+			log.error("Cannot initialize Velocity engine", e);
+			throw new APITemplateException("Cannot initialize Velocity engine", e);
+		}
+		VelocityContext context = configcontext.getContext();
 
-			Template t = velocityengine.getTemplate(this.getTemplatePath());
-
-			t.merge(context, writer);
+		return processTemplate(velocityengine, context, getVersionedWebAppTemplatePath());
+	}
 
+	@Override
+	public String getConfigStringForNonVersionedWebAppTemplate() throws APITemplateException {
+		VelocityEngine velocityengine = new VelocityEngine();
+		if (this.velocityLoggerName != null) {
+			velocityengine.setProperty(RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+									   "org.apache.velocity.runtime.log.Log4JLogChute");
+			velocityengine.setProperty("runtime.log.logsystem.log4j.logger", velocityLoggerName);
```

Comment:
Better to use "runtime.log.logsystem.log4j.logger" as a constant.
Formatting L113